compiler/javac/src/org/jetbrains/kotlin/wrappers/trees/JCMethod.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2010-2017 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jetbrains.kotlin.wrappers.trees
+
+import com.sun.source.util.TreePath
+import com.sun.tools.javac.tree.JCTree
+import org.jetbrains.kotlin.descriptors.Visibilities
+import org.jetbrains.kotlin.javac.Javac
+import org.jetbrains.kotlin.load.java.structure.JavaClass
+import org.jetbrains.kotlin.load.java.structure.JavaMethod
+import org.jetbrains.kotlin.load.java.structure.JavaType
+import org.jetbrains.kotlin.name.Name
+
+class JCMethod<out T : JCTree.JCMethodDecl>(tree: T,
+                                            treePath: TreePath,
+                                            containingClass: JavaClass,
+                                            javac: Javac) : JCMember<T>(tree, treePath, containingClass, javac), JavaMethod {
+
+    override val name
+        get() = Name.identifier(tree.name.toString())
```

Comment:
It cannot be done because `JCMember` type parameter is `<out T : JCTree>`. JCTree doesn't have a `name` field. At the same time `JCTree.JCMethodDecl` and `JCTree.JCVariableDecl` don't have a common ancestor class except for `JCTree`.